core/src/main/java/com/cloud/agent/api/routing/LoadBalancerConfigCommand.java

```diff
@@ -72,4 +82,35 @@ public NicTO getNic() {
     public Long getVpcId() {
         return vpcId;
     }
+
+    public LoadBalancerConfigTO[] getNetworkLbConfigs() {
+        return this.networkLbConfigs;
+    }
+
+    public void setNetworkLbConfigs(List<? extends LoadBalancerConfig> networkLbConfigs) {
+        if (networkLbConfigs == null) {
+            networkLbConfigs = new ArrayList<LoadBalancerConfig>();
+        }
+        this.networkLbConfigs = new LoadBalancerConfigTO[networkLbConfigs.size()];
```

Comment:
when `networkLbConfigs` is null, empty array is being created here. Is that the intention here? further, GC has to cleanup the reference for `networkLbConfigs`.

Reply:
> when `networkLbConfigs` is null, empty array is being created here. Is that the intention here? further, GC has to cleanup the reference for `networkLbConfigs`. same would be applicable for the similar code in this PR.

@sureshanaparti so your suggestion is ?

Reply:
@weizhouapache if you intention is to create an empty array, create `new LoadBalancerConfigTO[0]` and return when `networkLbConfigs` is null; else just return.

Reply:
@weizhouapache any update here?

Reply:
@sureshanaparti this has been changed in commit
https://github.com/apache/cloudstack/pull/4141/commits/1ccff8298166d66479d303b5be9196c23c8d0814

Reply:
> @sureshanaparti this has been changed in commit
> [1ccff82](https://github.com/apache/cloudstack/commit/1ccff8298166d66479d303b5be9196c23c8d0814)

thanks @weizhouapache verified.